sdks/python/apache_beam/runners/direct/direct_runner.py

```diff
@@ -86,6 +86,10 @@ def run(self, pipeline):
                         evaluation_context)
     # Start the executor. This is a non-blocking call, it will start the
     # execution in background threads and return.
+
+    from apache_beam.utils.value_provider import RuntimeValueProvider
+    if pipeline.options:
+      RuntimeValueProvider.set_runtime_options(pipeline.options._options_id, {})
```

Comment:
At the end of execution can you unset the runtime options only for DirectRunner? (Otherwise after each run there will accumulation of memory for previous PipelineOptions.)